zipline/pipeline/domain.py

```diff
@@ -0,0 +1,182 @@
+"""
+Domains
+-------
+
+TODO_SS
+"""
+from interface import implements, Interface
+
+from trading_calendars import get_calendar
+
+from zipline.country import CountryCode
+from zipline.utils.memoize import lazyval
+
+from .sentinels import NotSpecified
+
+
+class IDomain(Interface):
+    """Domain interface.
+    """
+
+    @property
+    def name(self):
+        """User-facing name for this domain.
+        """
+
+    @property
+    def country_code(self):
+        """Country code for assets on this domain.
+        """
+
+    # TODO_SS: The original design for domains was to have them return a
+    # TradingCalendar, but we have a bunch of tests in test_blaze that use very
+    # short session indices that I don't know how to port to using a proper
+    # TradingCalendar.
+    #
+    # Is there a strong reason to prefer just exposing the calendar
+    # vs. exposing the sessions? If so, what do we do about the blaze tests?
+    def get_sessions(self):
+        """Get all trading sessions for the calendar of this domain.
+        """
+
+
+# Create a base class so that we can do type-based validation of user input.
+Domain = implements(IDomain)
+Domain.__name__ = 'Domain'
+
+
+# TODO: Better name for this?
+# TODO: Do we want/need memoization for this?
+class StandardDomain(Domain):
+    """TODO_SS
+    """
+    def __init__(self, name, country_code, calendar_name):
+        self._name = name
+        self._country_code = country_code
+        self._calendar_name = calendar_name
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def country_code(self):
+        return self._country_code
+
+    def get_sessions(self):
+        return self.calendar.all_sessions
+
+    @lazyval
+    def calendar(self):
+        return get_calendar(self._calendar_name)
+
+    def __str__(self):
+        return self._name
+
+    def __repr__(self):
+        return "StandardDomain(name={!r}, country={!r}, calendar={!r})".format(
+            self._name,
+            self.country_code,
+            self._calendar_name,
+        )
+
+
+# TODO: Is this the casing convention we want for domains?
+USEquities = StandardDomain('USEquities', CountryCode.UNITED_STATES, 'NYSE')
```

Comment:
would it be helpful to namespace domains?

Reply:
nvm, realized you can just do `StandardDomain.__subclasses__()`

Reply:
I'm not sure we'd expect/want a user to be doing that, but the `domain` module itself is also potentially a perfectly good namespace.